include/ocpp/common/websocket/websocket.hpp

```diff
@@ -50,7 +54,7 @@ class Websocket {
 
     /// \brief register a \p callback that is called when the websocket connection has been closed and will not attempt
     /// to reconnect
-    void register_closed_callback(const std::function<void(const websocketpp::close::status::value reason)>& callback);
+    void register_closed_callback(const std::function<void(const WebsocketCloseReason reason)>& callback);
```

Comment:
We can keep the `closed_callback` and `disconnected_callback` for now because they are implicitly used to manage the reconnect behavior, but to reduce complexity of the interface I would prefer to not have the seperated after this refactor is done